docs/blog/2018-11-07-gatsby-for-apps/index.md

```diff
@@ -0,0 +1,249 @@
+
+To illustrate the concept, let's start with a classic example... we need to fetch some data at _run-time_ rather than build-time.
+
+```jsx:title=src/pages/messages.js
+import React from "react"
+
+import Layout from "../compoents/layout"
+
+class Messages extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      messages: [],
+    }
+  }
+
+  // note: this is a simplified example without error handling, authentication, etc.
+  async componentDidMount() {
+    const messages = await fetch(`/api/some-url-to-get-messages`).then(
+      response => response.json()
+    )
+
+    this.setState({
+      messages,
+    })
+  }
+
+  render() {
+    const { messages } = this.state
+    return (
+      <Layout>
+        {messages.length === 0 ? (
+          <p>Loading messages&hellip;</p>
+        ) : (
+          <ul>
+            {messages.map(message => (
+              <li key={message.id}>{message.text}</li>
+            ))}
+          </ul>
+        )}
+      </Layout>
+    )
+  }
+}
+
+export default Messages
+```
+
+All we've done here is implement the [`componentDidMount`][cdm] lifecycle method, which will fire off a request to some REST API to retrieve some data (messages!) from a remote API. At _runtime_ our application is now fetching data, dynamically. This illustrates the effectiveness of the Gatsby + React _runtime_ and how a Gatsby application is essentially a hydrated create-react-app like experience. [React lifecycle methods][react-lifecycle-methods] are fully supported to implement whatever dynamic interaction is required, like data fetching in this example!
+
+Consider the following animation, which mimics the end user experience. In effect, we've statically generated non-dynamic pieces (e.g. header, sidebar, etc.) and we then request additional data, on demand!
+
+![Loading dynamic data animation](./images/dynamic-data-fetching.gif)
+
+However--this is a fairly contrived, perhaps not completely real-world example. What if that REST API [requires authentication][authentication-tutorial]? What if we want [client-only routes][client-only-routes], e.g. deep linking to a specific message? You better believe it's possible! GraphQL at build time and run time? Yup!
+
+The central idea I want to clarify is that typically app-like features are not only _possible_ with Gatsby, but intuitive and easy to implement due to the dynamic runtime enabled in every Gatsby application. It's _just_ React.
+
+Gatsby is for building dynamic, web applications, just as it is for building static sites. At this point, this has become clear. Gatsby can be used for many of the traditional cases for web applications, including authentication, client-only routes, dynamic data fetching, and more.
+
+However, what hasn't been made clear is _why_ you'd reach for Gatsby as opposed to other solutions.
+
+## _Why_ Gatsby for Apps?
+
+If we revisit some of the benefits that Gatsby provides, minimally:
```

Comment:
I would give this its own subsection title -- it gets lost as an intro section, and it's really its own section within "Why Gatsby for Apps"